lib/hanami/view/application_context.rb

```diff
@@ -2,33 +2,56 @@
 
 module Hanami
   class View
-    module ApplicationContext
-      def initialize(inflector: Hanami.application.inflector, **options)
-        @inflector = inflector
-        super
-      end
+    class ApplicationContext < Module
+      attr_reader :provider
+      attr_reader :application
 
-      def inflector
-        @inflector
+      def initialize(provider)
+        @provider = provider
+        @application = provider.respond_to?(:application) ? provider.application : Hanami.application
       end
 
-      def request
-        _options.fetch(:request)
+      def included(context_class)
+        define_initialize
+        context_class.include(InstanceMethods)
       end
 
-      def session
-        request.session
+      private
+
+      def define_initialize
+        resolve_inflector = method(:resolve_inflector)
```

Comment:
```suggestion
        inflector = application.inflector
```

Might be worth keeping this simple and only moving to a method if the complexity grows?

Reply:
Makes sense, @timriley, done!!